biothings/web/query/pipeline.py

```diff
@@ -100,52 +106,47 @@ async def _(*args, **kwargs):
         ) as exc:
             logger.debug("QueryPipelineInterrupt: %s", exc.data)
             raise QueryPipelineInterrupt(exc.data)
-
-
         except RawResultInterrupt as exc:  # correspond to 'raw' option
             raise QueryPipelineInterrupt(exc.data.body)
-
-
         except AssertionError as exc:
             # in our application, AssertionError should be internal
             # the individual components raising the error should instead
-            # rasie exceptions like ValueError and TypeError for bad input
-            logging.exception("FIXME: Unexpected Assertion Error.", exc_info=exc)
+            # raise exceptions like ValueError and TypeError for bad input
+            logger.exception("FIXME: Unexpected Assertion Error.", exc_info=exc)
             raise QueryPipelineException(500, str(exc) or "N/A")
-
         except (ValueError, TypeError, ResultFormatterException) as exc:
             raise QueryPipelineException(400, type(exc).__name__, str(exc))
-
-        except elasticsearch.ConnectionError:  # like timeouts..
+        except ConnectionError:
             raise QueryPipelineException(503)
-
-        except elasticsearch.RequestError as exc:  # 400s
+        except RequestError as exc:
             raise QueryPipelineException(400, *_simplify_ES_exception(exc))
+        except NotFoundError as exc:
+            raise QueryPipelineException(404, *_simplify_ES_exception(exc))
+        except ConflictError as exc:
+            raise QueryPipelineException(409, *_simplify_ES_exception(exc))
+        except (AuthenticationException, AuthorizationException) as exc:
+            raise QueryPipelineException(403, *_simplify_ES_exception(exc))
+        except Exception as exc:
```

Comment:
We're doing a lot here in the general case. If possible I would try to add a little bit more context. For example in this chunk 
```
                if error_type == "search_phase_execution_exception":
                    if "rejected execution" in reason:
                        raise QueryPipelineException(503)
```
Why does `search_phase_execution_exception` coupled with `rejected_execution` translate to a 503 error